teammapper-frontend/mmp/src/map/handlers/nodes.ts

```diff
@@ -664,68 +689,141 @@ export default class Nodes {
+     * isRoot() - If parent node is root
+     * getOrientation() - Method to get the orientation of the node
+     * @returns 
      */
-    private calculateCoordinates(node: Node): Coordinates {
+    private calculateNodeCoordinates(
+        node: Node | ExportNodeProperties,
+        params: {
+            getParent: () => (Node | ExportNodeProperties) | null,
+            getSiblings: () => (Node | ExportNodeProperties)[],
+            isRoot: boolean,
+            getOrientation: (n: Node | ExportNodeProperties) => boolean | undefined
+        }
+    ): Coordinates {
+        const nodeParent = params.getParent();
+        
         let coordinates: Coordinates = {
-            x: node.parent ? node.parent.coordinates.x : node.coordinates.x || 0,
-            y: node.parent ? node.parent.coordinates.y : node.coordinates.y || 0
-        },
-            siblings: Array<Node> = this.getSiblings(node)
-
-        if (node.parent && node.parent.isRoot) {
-            const rightNodes: Array<Node> = [],
-                leftNodes: Array<Node> = []
-
-            for (const sibling of siblings) {
-                this.getOrientation(sibling) ? leftNodes.push(sibling) : rightNodes.push(sibling)
-            }
-
+            x: nodeParent?.coordinates?.x ?? (node.coordinates?.x ?? 0),
+            y: nodeParent?.coordinates?.y ?? (node.coordinates?.y ?? 0)
+        };
+    
+        let siblings = params.getSiblings();
+    
+        if (nodeParent && params.isRoot) {
+            const [leftNodes, rightNodes] = siblings.reduce<[(Node | ExportNodeProperties)[], (Node | ExportNodeProperties)[]]>(
+                (acc, sibling) => {
+                    params.getOrientation(sibling) 
+                        ? acc[0].push(sibling) 
+                        : acc[1].push(sibling);
+                    return acc;
+                },
+                [[], []]
+            );
+    
             if (leftNodes.length <= rightNodes.length) {
-                coordinates.x -= 200
-                siblings = leftNodes
+                coordinates.x -= NODE_HORIZONTAL_SPACING;
+                siblings = leftNodes;
             } else {
-                coordinates.x += 200
-                siblings = rightNodes
+                coordinates.x += NODE_HORIZONTAL_SPACING;
+                siblings = rightNodes;
             }
         } else if (!node.detached) {
-            if (node.parent && this.getOrientation(node.parent)) {
-                coordinates.x -= 200
+            if (nodeParent && params.getOrientation(nodeParent)) {
+                coordinates.x -= NODE_HORIZONTAL_SPACING;
             } else {
-                coordinates.x += 200
+                coordinates.x += NODE_HORIZONTAL_SPACING;
             }
         }
-
+    
         if (siblings.length > 0) {
-            const lowerNode = this.getLowerNode(siblings)
-            coordinates.y = lowerNode.coordinates.y + 60
+            const lowerNode = 'isRoot' in node 
```

Comment:
and `isRoot === true` ? Is it enough that the property only exists, but being false?

Reply:
Check has been removed, unnecessarily complex.